blender/arm/logicnode/logic_gate.py

```diff
@@ -3,6 +3,13 @@
 from bpy.types import Node, NodeSocket
 from arm.logicnode.arm_nodes import *
 
+def remove_extra_inputs(self, context):
+    # if not any(p == self.property0 for p in ['Or', 'And']):
+    #     if len(self.inputs) > self.min_inputs:
+    #         #FIXME: str(id(self)) gives us a key error here
```

Comment:
I'm not super familiar with Blender's API. If someone more experienced could help figure out how to clean up the extra input sockets when switching to a non-And/Or mode that'd be much appreciated.